comm/src/main/scala/coop/rchain/p2p/p2p.scala

```diff
@@ -63,30 +65,32 @@ final case class Network(
     * allowing encryption for all future messages. Next protocols are agreed on to ensure that these two nodes can speak
     * the same language.
     */
-  def connect(peer: PeerNode): Unit = {
-    logger.debug(s"connect(): Connecting to $peer")
-    val ehs = EncryptionHandshakeMessage(
-      encryptionHandshake(net.local, keys),
-      System.currentTimeMillis,
-    )
-    val remote = new ProtocolNode(peer, this.net)
-    net.roundTrip(ehs, remote) match {
-      case Right(resp) => {
-        logger.debug(s"connect(): Received encryption handshake response from ${resp.sender.get}.")
-        val phs = ProtocolHandshakeMessage(NetworkProtocol.protocolHandshake(net.local),
-                                           System.currentTimeMillis)
-        net.roundTrip(phs, remote) match {
-          case Right(resp) => {
-            logger.debug(
-              s"connect(): Received protocol handshake response from ${resp.sender.get}.")
-            net.add(remote)
+  def connect[F[_]: Capture: FlatMap](peer: PeerNode): F[Unit] =
+    for {
+      _  <- iologger.debug[F](s"connect(): Connecting to $peer")
+      ts <- IOUtil.currentMilis[F]
+      proto  = encryptionHandshake(net.local, keys)
+      ehs    = EncryptionHandshakeMessage(proto, ts)
+      remote = new ProtocolNode(peer, this.net)
+    } yield {
+      net.roundTrip(ehs, remote) match {
+        case Right(resp) => {
+          logger.debug(
+            s"connect(): Received encryption handshake response from ${resp.sender.get}.")
+          val phs = ProtocolHandshakeMessage(NetworkProtocol.protocolHandshake(net.local),
+                                             System.currentTimeMillis)
```

Comment:
Use IOUtils.currentMilis here?

Reply:
Yes, but not yet. As you see `rountTrip` is NOT yet refactored to use IO, thus we still yield a bit of effectful code (next PR will change it and this bit (as well as logger) will be refactored)